### PR TITLE
feat: add support for WinUI and impl WebView2 for WinUI package

### PR DESCRIPTION
**Pull request by GabrielChouinardLetourneau**
GitHub Issue: #

## What is the current behavior?

No support for WinUI and using the WebView from Windows.UI.Xaml.

## What is the new behavior?

Add support for WinUI and implement WebView2 for that new package

## Checklist

Please check if your PR fulfills the following requirements:

- [ ] Documentation has been added/updated
- [ ] Automated Unit / Integration tests for the changes have been added/updated
- [ ] Contains **NO** breaking changes
- [ ] Updated the Changelog
- [ ] Associated with an issue

### CODE REVIEW

File: src/AsyncWebView.Uno.WinUI/AsyncWebView/AsyncWebView.WinUI.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
+﻿#if WINDOWS10_0_18362_0_OR_GREATER
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.Web.WebView2.Core;
+using Windows.Web.Http.Filters;
+
+namespace AsyncWebView
```
Comment (carlh98) [inline src/AsyncWebView.Uno.WinUI/AsyncWebView/AsyncWebView.WinUI.cs:10]: namespace AsyncWebView;

File: src/AsyncWebView.Uno.WinUI/AsyncWebView/AsyncWebView.WinUI.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+{
+	/// <summary>
+	/// Implementation of <see cref="AsyncWebView"/> for Windows
```
Comment (carlh98) [inline src/AsyncWebView.Uno.WinUI/AsyncWebView/AsyncWebView.WinUI.cs:13]: dot at the end

File: src/AsyncWebView.Uno/AsyncWebView.Uno.csproj [hunk 3]
```diff
@@ -12,16 +12,22 @@
...
+		<PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
+		<PackageProjectUrl>https://github.com/nventive/AsyncWebView</PackageProjectUrl>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(TargetFramework)'=='netstandard2.0'">
     <DefineConstants>$(DefineConstants);__WASM__</DefineConstants>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Uno.UI" Version="4.9.0-dev.811" />
```
Comment (carlh98) [inline src/AsyncWebView.Uno/AsyncWebView.Uno.csproj:22]: Why do we increase the version of Uno.UI?

Comment (carlh98) [reply src/AsyncWebView.Uno/AsyncWebView.Uno.csproj:22]: It should stay at the main branch version

File: src/AsyncWebView.Uno.WinUI/AsyncWebView/AsyncWebView.cs [hunk 4]
```diff
@@ -0,0 +1,630 @@
+﻿#if WINDOWS || __ANDROID__ || __IOS__ || __WASM__
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Uno.Logging;
+using Windows.System;
+using System.Net.Http;
```
Comment (carlh98) [inline src/AsyncWebView.Uno.WinUI/AsyncWebView/AsyncWebView.cs:8]: reorder the usings

File: src/AsyncWebView.Uno.WinUI/AsyncWebView/AsyncWebView.cs [hunk 5]
```diff
@@ -0,0 +1,630 @@
...
+using Microsoft.Extensions.Logging.Abstractions;
+using System.Linq;
+using Uno.Disposables;
+using _WebView = Microsoft.UI.Xaml.Controls.WebView2;
+using NavigationStartingEventArgs = Microsoft.Web.WebView2.Core.CoreWebView2NavigationStartingEventArgs;
+using NavigationCompletedEventArgs = Microsoft.Web.WebView2.Core.CoreWebView2NavigationCompletedEventArgs;
+using NavigationFailedEventArgs = Microsoft.Web.WebView2.Core.CoreWebView2ProcessFailedEventArgs;
+using ScriptDialogOpeningEventArgs = Microsoft.Web.WebView2.Core.CoreWebView2ScriptDialogOpeningEventArgs;
+
+namespace AsyncWebView
```
Comment (carlh98) [inline src/AsyncWebView.Uno.WinUI/AsyncWebView/AsyncWebView.cs:24]: namespace AsyncWebView;

**Comment by GabrielChouinardLetourneau** [comment]
I will push changes related to the pipeline

**Review by Guidemarcus — CHANGES_REQUESTED** [review]

File: src/AsyncWebView.Uno/AsyncWebView.Uno.csproj [hunk 6]
```diff
@@ -12,16 +12,22 @@
...
-		<PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.9.0-dev.811" />
+    <PackageReference Include="Uno.UI" Version="4.9.0-dev.969" />
```
[the remark this answers is not in the document]
Comment (GabrielChouinardLetourneau) [reply src/AsyncWebView.Uno/AsyncWebView.Uno.csproj]: This is update in the feature branch in the last [PR](https://github.com/nventive/AsyncWebView/pull/28), but I reverted the change like CH mentioned

File: src/AsyncWebView.Uno.WinUI/AsyncWebView.Uno.WinUI.csproj [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    <PackageReference Include="Uno.WinUI" Version="4.9.0-dev.969" />
+		<PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="5.0.0" />
+		<PackageReference Include="Uno.Core" Version="4.0.1" />
+		<PackageReference Include="Uno.Core.Extensions.Logging" Version="4.0.1" />
+  </ItemGroup>
+  <Choose>
+    <When Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
+      <ItemGroup>
+        <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.2.230313.1" />
+        <PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.755" />
```
[the remark this answers is not in the document]
Comment (GabrielChouinardLetourneau) [reply src/AsyncWebView.Uno.WinUI/AsyncWebView.Uno.WinUI.csproj:38]: Yes, we need them for the windows net 6 target

File: src/Shared/Helpers/CompletionCommandArgs.cs [hunk 8]
```diff
@@ -1,4 +1,4 @@
-﻿#if WINDOWS_UWP || __ANDROID__ || __IOS__ || __WASM__
+﻿#if WINDOWS_UWP || __ANDROID__ || __IOS__ || __WASM__ || WINDOWS10_0_18362_0_OR_GREATER
```
[the remark this answers is not in the document]
Comment (GabrielChouinardLetourneau) [reply src/Shared/Helpers/CompletionCommandArgs.cs:1]: I will define a constant WINDOWS_WINUI to make it more readable. Nventive.View had the same strategy

File: src/AsyncWebView.Uno.WinUI/AsyncWebView.Uno.WinUI.csproj [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+		<!-- Needed for Source Link support -->
+		<PublishRepositoryUrl>true</PublishRepositoryUrl>
+		<EmbedUntrackedSources>true</EmbedUntrackedSources>
+		<IncludeSymbols>true</IncludeSymbols>
+		<SymbolPackageFormat>snupkg</SymbolPackageFormat>
+  </PropertyGroup>
+	<PropertyGroup Condition="'$(TargetFramework)'=='netstandard2.0'">
+		<DefineConstants>$(DefineConstants);__WASM__</DefineConstants>
+	</PropertyGroup>
```
[the remark this answers is not in the document]
Comment (GabrielChouinardLetourneau) [reply src/AsyncWebView.Uno.WinUI/AsyncWebView.Uno.WinUI.csproj:27]: We need that constant for netstandard and 1 condition regarding the `StringExtensions`, unless we remove the netstandard target, I would keep it like other WinUI packages we created

**Review by Guidemarcus — APPROVED** [review]

File: src/AsyncWebView.Uno/AsyncWebView.Uno.csproj [hunk 10]
```diff
@@ -12,16 +12,22 @@
...
   <PropertyGroup Condition="'$(TargetFramework)'=='netstandard2.0'">
     <DefineConstants>$(DefineConstants);__WASM__</DefineConstants>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Uno.UI" Version="4.9.0-dev.811" />
-		<PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.9.0-dev.811" />
+    <PackageReference Include="Uno.UI" Version="4.0.7" />
+		<PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.0.7" />
+		<PackageReference Include="Uno.SourceGenerationTasks" Version="4.2.0" />
```
Comment (carlh98) [inline src/AsyncWebView.Uno/AsyncWebView.Uno.csproj:27]: this is needed?

**Pull request merged**